src/identity_component.cairo

```diff
@@ -121,20 +121,18 @@ pub mod IdentityComponent {
             key_type: felt252,
         ) -> bool {
             self.only_manager();
-            let mut key_storage_path = self.Identity_keys.entry(key);
-            if key_storage_path.key.read() == key {
-                let purposes_storage_path = key_storage_path.purposes.deref();
-                for i in 0..purposes_storage_path.len() {
-                    assert(
-                        purpose != purposes_storage_path[i].read(), Errors::KEY_ALREADY_HAS_PURPOSE,
-                    );
-                };
-                purposes_storage_path.append().write(purpose);
-            } else {
-                key_storage_path.key.write(key);
-                key_storage_path.key_type.write(key_type);
-                key_storage_path.purposes.deref().append().write(purpose);
-            }
+            let key_storage_path = self.Identity_keys.entry(key);
+            let mut key_details = key_storage_path.read();
+            let purpose_bit_index = purpose.try_into().expect('Invalid Purpose');
+            assert(purpose_bit_index < 128, 'Purpose is not in valid range');
```

Comment:
128 seems to come magically here. Consider moving this assertion inside the Bitmap implementation.